src/common/mfu_daos.c

```diff
@@ -992,10 +993,32 @@ int daos_connect(
     int             valid = 0; /* assume failure until otherwise */
     int             rc;
     daos_prop_t*    props = NULL;
+    struct daos_prop_entry *entry;
+    char *filename;
+#ifdef HDF5_SUPPORT
+    struct hdf5_args hdf5;
+#endif
 
     /* have rank 0 connect to the pool and container,
      * we'll then broadcast the handle ids from rank 0 to everyone else */
     if (rank == 0) {
+#ifdef HDF5_SUPPORT
+        /* initialization for hdf5 file for preserve option */
+        if (preserve) {
+            rc = asprintf(&filename, "%s%s%s", da->daos_preserve_path,
+                          "/", "daos_metadata.hdf5");
```

Comment:
@dsikich Maybe we should allow the user to pass in the path to the file itself. I can see how users might have several of these metadata files for different scenarios, and they might want to pick a specific one.

Reply:
@daltonbohning I went back and forth on this, and wasn't sure if I should just create the name of the file or expect the user to pass in a name. I think what you suggested below is good where we only create the name if the user passes in a directory.